### PR TITLE
(PUP-7042) Mark strings in provider

### PR DESCRIPTION
**Pull request by Magisus**
This commit marks user-facing error and info strings in
`lib/puppet/provider.rb` and `lib/puppet/provider/*` for translation.

### CODE REVIEW

**Comment by Magisus** [comment]
Closing and reopening to fix Travis.

**Comment by Magisus** [comment]
Closing and reopening to fix CI.

File: lib/puppet/provider/mount/parsed.rb [hunk 1]
```diff
@@ -123,11 +123,11 @@ def post_parse(result)
         result[:options] = [result[:options],special_options.sort].flatten.compact.join(',')
         if ! result[:device]
           result[:device] = :absent
-          Puppet.err "Prefetch: Mount[#{result[:name]}]: Field 'device' is missing"
+          Puppet.err _("Prefetch: Mount[%{name}]: Field 'device' is missing") % { name: result[:name] }
```
Comment (MosesMendoza) [inline lib/puppet/provider/mount/parsed.rb]: Can you add a note that `Prefetch` is a program name here? Same for next externalization

File: lib/puppet/provider/network_device.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ def self.prefetch(resources)
     end
   rescue => detail
     # Preserving behavior introduced in #6907
-    Puppet.log_exception(detail, "Could not perform network device prefetch: #{detail}")
+    Puppet.log_exception(detail, _("Could not perform network device prefetch: %{detail}") % { detail: detail })
```
Comment (MosesMendoza) [inline lib/puppet/provider/network_device.rb]: same re: prefetch

**Comment by Magisus** [comment]
Updated.

**Pull request merged**